cpp/template/CMakeLists.txt

```diff
@@ -36,3 +36,8 @@ include(cmake/thirdparty/get_raft.cmake)
 # -------------- compile tasks ----------------- #
 add_executable(TEST_RAFT src/test_vector_search.cu)
 target_link_libraries(TEST_RAFT PRIVATE raft::raft raft::compiled)
+target_compile_definitions(TEST_RAFT PRIVATE "RAFT_EXPLICIT_INSTANTIATE_ONLY")
+
+add_executable(IVF_FLAT_EXAMPLE src/ivf_flat_example.cu)
+target_link_libraries(IVF_FLAT_EXAMPLE PRIVATE raft::raft raft::compiled)
+target_compile_definitions(IVF_FLAT_EXAMPLE PRIVATE "RAFT_EXPLICIT_INSTANTIATE_ONLY")
```

Comment:
I think we should pass an option back to the main build target to set this value on `raft::raft` target, which should get passed through transitively. It's an extra step that is, quite frankly, not self explanatory and confusing to see. Users should just be able to include their `get_raft.cmake` and add the needed raft targets.

Reply:
I don't mean passing it through implicitly by default, I mean that we should provide a cmake option that users can set so that the flag is set the raft::raft target by the raft cmake. It follows the standardized way that we enable such options.

Reply:
I would treat this separately from the current PR (especially that I do not know what is the right way to achieve this in cmake). I have created issue #1840. For the current PR, I have removed the flag (and comments) from CMakelist.txt.